provider/hybrik/hybrik.go

```diff
@@ -224,18 +226,21 @@ func (hp *hybrikProvider) presetsToTranscodeJob(job *db.Job) (string, error) {
 		var segmentDur uint
 		// track the hls outputs so we can later connect them to a manifest creator task
 		if len(preset.Payload.Targets) > 0 && preset.Payload.Targets[0].Container.Kind == hls {
-			hlsElementIds = append(hlsElementIds, len(elements))
+			hlsElementIds = append(hlsElementIds, len(transcodeElementIds))
 			segmentDur = job.StreamingParams.SegmentDuration
 		}
 
-		e := hp.mountTranscodeElement(strconv.Itoa(len(elements)), job.ID, output.FileName, hp.config.Destination, segmentDur, preset)
+		taskIndex := strconv.Itoa(len(transcodeElementIds))
+		e := hp.mountTranscodeElement(taskIndex, job.ID, output.FileName, hp.config.Destination, segmentDur, preset)
+
+		transcodeElementIds = append(transcodeElementIds, fmt.Sprintf(transcodeElementIDTempate, taskIndex))
 		elements = append(elements, e)
 	}
 
 	// connect the source element to each of the transcode elements
-	transcodeSuccessConnections := make([]hwrapper.ToSuccess, len(elements))
-	for i := range elements {
-		transcodeSuccessConnections[i] = hwrapper.ToSuccess{Element: "transcode_task" + strconv.Itoa(i)}
+	transcodeSuccessConnections := make([]hwrapper.ToSuccess, len(transcodeElementIds))
+	for i, id := range transcodeElementIds {
+		transcodeSuccessConnections[i] = hwrapper.ToSuccess{Element: id}
```

Comment:
Could we range through elements and add element.UID to the transcodeSuccessConnections slice? Something like:

```go
for i, e := range elements {
  transcodeSuccessConnections[i] = hwrapper.ToSuccess{Element: e.UID}
}
```

Then we don't need an additional slice, I think. And that would also keep the logic to generate the element id inside the monutTranscodeElement function.

If we do that, can you also make the constant ``transcodeElementIDTemplate`` local to that function?

Reply:
This change was made because `elements` holds a slice of *all* elements and in this case, not only the transcode tasks elements, but at least one source element created and appended before we run the logic to create the transcode task elements. (this was causing incorrect connections to be generated)

Rather than assume we have a single source element and doing a range over `elements[1:]` (along with assuming there are no elements between the source and the transcode tasks), I thought it would be less error prone to explicitly track transcode tasks. This allows us to freely add elements before the transcode elements without having to change that slice offset.

We can safely change 
`transcodeElementIds = append(transcodeElementIds, fmt.Sprintf(transcodeElementIDTemplate, taskIndex))`
to
`transcodeElementIds = append(transcodeElementIds, e.UID)`

but we make use of `transcodeElementIDTemplate` again in `presetsToTranscodeJob` when generating connections to packagings tasks (line 293)

Reply:
Thanks for clarifying! This makes sense, my bad x)

We can keep the code as is :D

Reply:
Cheers! I was tripped up by the code a few times myself, I think we can clean it up a little :)